src/Workspaces/Core/Portable/Execution/AbstractOptionsSerializationService.cs

```diff
@@ -250,6 +251,7 @@ protected OptionSet ReadOptionSetFrom(OptionSet options, string language, Object
             options = ReadOptionFrom(options, language, CodeStyleOptions.PreferNullPropagation, reader, cancellationToken);
             options = ReadOptionFrom(options, language, CodeStyleOptions.PreferObjectInitializer, reader, cancellationToken);
             options = ReadOptionFrom(options, language, CodeStyleOptions.PreferThrowExpression, reader, cancellationToken);
+            options = ReadOptionFrom(options, language, CodeStyleOptions.PreferReadonly, reader, cancellationToken);
```

Comment:
this is the wrong order.  you are writing the option last, but reading it second to last.